controllers/cartController.js

```diff
@@ -0,0 +1,19 @@
+const cartService = require("../services/cartService");
+const { catchAsync, raiseCustomError } = require("../utils/error");
+
+const addProduct = catchAsync(async (req, res) => {
+  const user = req.user;
+  const userId = user.id;
```

Comment:
`user` 변수는 선언할 필요가 없어보이네요.
```suggestion
  const userId = req.user.id;
```